Utilities/MachineInventory.cs

```diff
@@ -0,0 +1,245 @@
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+namespace FortressCraft.Community.Utilities
+{
+    /// <summary>
+    ///     Generic machine inventory class with list inventory support
+    /// </summary>
+    public class MachineInventory
```

Comment:
Class looks helpful - is this supposed to act as the delegate for machines that have inventory? I would suggest making null checks in all of your functions for safety.